src/main/java/com/example/controllers/TransferController.java

```diff
@@ -0,0 +1,145 @@
+
+		return "newTransfer";
+	}
+
+	@RequestMapping(method = RequestMethod.POST)
+	public String transfer(@Valid @ModelAttribute final Transfer transfer, final BindingResult bindingResult, final Model model,
+			final Principal principal, @CookieValue(value = "accountType", defaultValue = AccountType.PERSONAL) final String accountType,
+			final HttpSession session, final HttpServletResponse response) {
+
+		if (bindingResult.hasErrors()) {
+			return newTransferForm(model, principal, response);
+		}
+		else {
+			return AccountType.PERSONAL.equals(accountType) ? transferCheck(transfer, model, session, principal)
+					: transferConfirmation(transfer, model, principal, accountType);
+		}
+	}
+
+	private String transferCheck(final Transfer transfer, final Model model, final HttpSession session, final Principal principal) {
+		Account account = accountDao.findUsersByUsername(principal.getName()).get(0);
+		model.addAttribute("account", account);
+		model.addAttribute("transferbean", transfer);
+		model.addAttribute("operationConfirm", new OperationConfirm());
+		session.setAttribute(PENDING_TRANSFER, transfer);
+
+		return "transferCheck";
+	}
+
+	private String transferConfirmation(final Transfer transfer, final Model model, final Principal principal, final String accountType) {
+
+		List<CashAccount> cashAccounts = cashaccountDao.findCashAccountsByUsername(principal.getName());
+		Account account = accountDao.findUsersByUsername(principal.getName()).get(0);
+		double aux = transfer.getAmount();
+
+		if (aux == 0.0) {
+
+			model.addAttribute("transfer", transfer);
+			model.addAttribute("cashAccounts", cashAccounts);
+			model.addAttribute("account", account);
+			model.addAttribute("error", true);
+
+			return "newTransfer";
+		}
+
+		transfer.setUsername(principal.getName());
+		transfer.setDate(new Date());
+
+		double amount = transfer.getAmount();
+		transfer.setAmount(InsecureBankUtils.round(amount, 2));
+
+		double feeAmount = transfer.getAmount() * transfer.getFee() / 100.0;
+		transfer.setFee(InsecureBankUtils.round(feeAmount, 2));
+
+		transfersFacade.createNewTransfer(transfer);
+
+		model.addAttribute("transferbean", transfer);
+		model.addAttribute("account", account);
+		model.addAttribute("accountType", accountType);
+
+		return "transferConfirmation";
+	}
+
+	@RequestMapping(value = "/confirm", method = RequestMethod.POST)
+	public String transferCheck(final OperationConfirm operationConfirm, final BindingResult bindingResult, final Model model,
+			final Principal principal, @CookieValue(value = "accountType", defaultValue = AccountType.PERSONAL) final String accountType,
+			final HttpSession session) {
+
+		Transfer transfer = (Transfer) session.getAttribute(PENDING_TRANSFER);
+		session.removeAttribute(PENDING_TRANSFER);
+
+		if ("confirm".equals(operationConfirm.getAction()) && transfer != null) {
+			return transferConfirmation(transfer, model, principal, accountType); 
```

Comment:
#### Automated PR Comment From Polaris SAST
<!-- eacc809524aa4fa308a02b1bcf3d5a20 -->
<!-- Comment managed by Git Commenter, do not modify! -->
### Polaris SAST Issue - SQL Injection
_High CWE-89_
Untrusted user-supplied data is inserted into a SQL statement without adequate validation, escaping, or filtering.

A user can change the intent of the SQL query, which may inappropriately disclose or corrupt data within the database.

### How to fix
Rewrite all SQL queries constructed through dynamic concatenation to use an injection-safe query mechanism such as prepared statements with parameterized queries.

Most modern programming languages provide a feature called "parameterized queries" that allow user-supplied data to be inserted safely as values in dynamic SQL queries. Rather than construct the dynamic SQL query by concatenating user-supplied data to static SQL query string fragments, data values are identified in the query by parameter markers or variables. Dynamic data is then passed through a mechanism provided by SQL that prevents the supplied data from changing the meaning of the query.

Note: the exact syntax and use of prepared statements with parameterized queries vary from language to language.